### PR TITLE
feat(api): add submit result endpoint

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This pull request intends to add 3 new endpoints to the API. Those endpoints can be used to submit a result to resources, host and service.

**Endpoints:**
- **POST** `/monitoring/resources/submit`
- **POST** `/monitoring/hosts/<hostId>/submit`
- **POST** `/monitoring/hosts/<hostId>/services/<serviceId>/submit`

**Example**

If I want to submit a result to several resources here is an example of the body that I should provide

```
      {
        "resources": [
          {
            "type": "host",
            "id": 19,
            "parent": null,
            "status": 2,
            "output": "Host DOWN",
            "performance_data": "nbproc: 0"
          },
          {
            "type": "service",
            "id": 135,
            "parent": {
              "id": 19
            },
            "status": 2,
            "output": "Service CRITICAL",
            "performance_data": "nbproc: 0"
          }
        ]
      }
```

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

Follow the added documentation to test the endpoints.
**Do not forget to test the endpoints using an user with limited rights**

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by sc979** [comment]
@adr-mo , I removed the centreon-php reviewer, until you set this PR to ready to review ;)

File: config/json_validator/latest/Centreon/SubmitResult/SubmitResult.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        "performance_data": {
+            "type": [
+                "string",
+                "null"
+            ]
+        },
+        "output": {
+            "type": ["string",
+                "null"
+            ]
```
Comment (kduret) [inline config/json_validator/latest/Centreon/SubmitResult/SubmitResult.json:28]:
```suggestion
            "type": [
                "string",
                "null"
            ]
```

File: src/Centreon/Application/Controller/Monitoring/SubmitResultController.php [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new SubmitResultException($message);
+        }
+
+        $status = ResourceStatus::STATUS_MAPPING[strtoupper($results['status'])];
```
Comment (kduret) [inline src/Centreon/Application/Controller/Monitoring/SubmitResultController.php:260]: all this section seems exactly the same within `submitResultHost` method
you should extract this in a separated private method

File: tests/php/Centreon/Domain/Engine/EngineServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            $this->service->getHost()->getName(),
+            $this->service->getDescription(),
+            $this->serviceResult->getStatus(),
+            $this->serviceResult->getOutput(),
+            $this->serviceResult->getPerformanceData()
+        );
+
+        $this->engineRepository->expects($this->once())
+            ->method('sendExternalCommand')
+            ->with($this->equalTo($this->commandHeader . $command));
```
Comment (kduret) [inline tests/php/Centreon/Domain/Engine/EngineServiceTest.php:154]: Nice !

File: src/Centreon/Domain/Contact/Contact.php [hunk 4]
```diff
@@ -444,6 +446,7 @@ public function setAccessToApiRealTime(bool $hasAccessToApiRealTime): self
      */
     public function hasRole(string $role): bool
     {
+        $roles = $this->$roles;
```
Comment (kduret) [inline src/Centreon/Domain/Contact/Contact.php:449]:
```suggestion
```

**Review by callapa — APPROVED** [review]

**Review by callapa — CHANGES_REQUESTED** [review]
Please commit the Kevin's suggestion

File: src/Centreon/Domain/Contact/Contact.php (hunk 4 above)
Comment (adr-mo) [reply src/Centreon/Domain/Contact/Contact.php:449]: Yes sorry it was part of my debugging :S

**Review by callapa — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**